src/gleam/httpc.gleam

```diff
@@ -98,6 +145,124 @@ pub fn send_bits(
   |> dispatch_bits(req)
 }
 
+/// Send a HTTP stream request of binary data
+/// 
+pub fn dispatch_stream_bits(
+  config: Configuration,
+  req: Request(BitArray),
+) -> Result(RequestIdentifier, HttpError) {
+  let erl_url =
+    req
+    |> request.to_uri
+    |> uri.to_string
+    |> charlist.from_string
+  let erl_headers = prepare_headers(req.headers)
+  let erl_http_options = [
+    Autoredirect(config.follow_redirects),
+    Timeout(config.timeout),
+  ]
+  let erl_http_options = case config.verify_tls {
+    True -> erl_http_options
+    False -> [Ssl([Verify(VerifyNone)]), ..erl_http_options]
+  }
+  let erl_options = [
+    BodyFormat(Binary),
+    SocketOpts([Ipfamily(Inet6fb4)]),
+    Sync(False),
+    Stream(#(Self, Once)),
+  ]
+  use request_id <- result.try(
+    case req.method {
+      http.Options | http.Head | http.Get -> {
+        let erl_req = #(erl_url, erl_headers)
+        erl_stream_request_no_body(
+          req.method,
+          erl_req,
+          erl_http_options,
+          erl_options,
+        )
+      }
+      _ -> {
+        let erl_content_type =
+          req
+          |> request.get_header("content-type")
+          |> result.unwrap("application/octet-stream")
+          |> charlist.from_string
+        let erl_req = #(erl_url, erl_headers, erl_content_type, req.body)
+        erl_stream_request(req.method, erl_req, erl_http_options, erl_options)
+      }
+    }
+    |> result.map_error(normalise_error),
+  )
+
+  Ok(request_id)
+}
+
+/// Triggers the next asynchronous streaming message to be sent to the calling process
```

Comment:
Wrap before 80 columns please 🙏 

Could we use a name more like the name in the other library please